### ISSUE TITLE
Why does it sometimes not take effect at all after being enabled？

### ISSUE THREAD

**Issue by Matrixgo** [ototadana/sd-face-editor#172]
I am using img2img, as you can see, I have enabled sd-face-editor. 

Sometimes sd-face-editor runs well, but sometimes it doesn't work at all. I have been trying for a long time and I don't understand what caused it. Is there anyone else in the same situation as me? Do you have any ideas? 

![20231121-200558](https://github.com/ototadana/sd-face-editor/assets/41765932/b6106184-8f82-4eb0-b90b-d5cd58f4c350)

**Comment by ototadana** [comment]
The face detector used in the default workflow often fails to detect anime faces.

In such cases, using [lbpcascade_animeface](https://github.com/ototadana/sd-face-editor/blob/main/workflows/examples/README.md#example-4-anime-face-detection-workflow---lbpcascade_animeface-example) will increase the probability that faces will be detected.

To use this, please apply the following settings in the `Face Editor` section of the `Settings` tab and restart the Web UI.

- Enable "Search workflows in subdirectories".
![image](https://github.com/ototadana/sd-face-editor/assets/4137885/af7a0853-e99d-40b1-9956-66f63f09a7aa)

Then select "examples/lbpcascade_animeface" to generate images.
![image](https://github.com/ototadana/sd-face-editor/assets/4137885/de23393e-9a8b-4870-abd6-76df31a90faf)

**Comment by Matrixgo** [comment]
@ototadana Thank you very much for your detailed answer. It has really helped me and looks like the effect is good! 

But now there is another question, how can I choose to use this new workflow in the API parameters so that it can take effect directly like on the web UI?

![20231122-104107](https://github.com/ototadana/sd-face-editor/assets/41765932/847aed12-1e52-4410-a39e-c2a66eb02274)

**Comment by ototadana** [comment]
@Matrixgo 
I am glad to hear that it worked.

In the API, use the "workflow" parameter. The value of the parameter should be an escaped string of [JSON that defines the workflow](https://github.com/ototadana/sd-face-editor/blob/main/workflows/examples/lbpcascade_animeface.json).

For example: 
```json
{
    "prompt": "a girl with wings",
    "steps": 20,
    "alwayson_scripts": {
      "face editor ex": {
         "args": [{"workflow": "{\n \"face_detector\": \"lbpcascade_animeface\",\n \"rules\": {\n \"then\": {\n \"face_processor\": \"img2img\",\n \"mask_generator\": {\n \"name\": \"BiSeNet\",\n \"params\": {\n \"fallback_ratio\": 0.3\n }\n }\n }\n }\n}"}]
      }
    }
}
```

**Comment by Matrixgo** [comment]
@ototadana 
Thank you! I really feel like your reply. I tried calling the API according to your example, but it doesn't seem to fix the face correctly as on the interface. My key parameters are as follows. Is it my incorrect usage?

>http://127.0.0.1:7860/sdapi/v1/img2img

`{
    "init_images": ["xxxxx"],
    "prompt": "xxxxx",
    "negative_prompt": "xxxxx",
    "override_settings": {
        "sd_model_checkpoint": "xxxxx",
        "CLIP_stop_at_last_layers":2,
		"eta_noise_seed_delta":31337
    },
    "eta":0.667,
    "steps": 25,
    "sampler_name": "Euler a",
    "cfg_scale": 7,
    "seed": -1,
    "width": 512,
    "height": 720,
    "denoising_strength": 0.5,
    "alwayson_scripts": {
        "face editor ex": {
            "args": [
                {
                    "enabled": true,
                    "workflow": "{\n \"face_detector\": \"lbpcascade_animeface\",\n \"rules\": {\n \"then\": {\n \"face_processor\": \"img2img\",\n \"mask_generator\": {\n \"name\": \"BiSeNet\",\n \"params\": {\n \"fallback_ratio\": 0.3\n }\n }\n }\n }\n}"
                }
            ]
        },
        "controlnet": {
            "args": [
                {
                    "enabled": true,
                    "module": "canny",
                    "model": "control_v11p_sd15_canny [d14c016b]",
                    "control_mode": 1,
                    "pixel_perfect": true
                }
            ]
        }
    }
}`

**Comment by ototadana** [comment]
@Matrixgo 
I have written the following program using the values you provided to check its behavior, and it is working properly without any problems.

Note that "prompt_for_face" and "show_intermediate_steps" have been added so that we can clearly determine if it is working properly.

```python

import base64
import io

import cv2
import requests
from PIL import Image

url = "http://127.0.0.1:7860"

img = cv2.imread("input-01.png")[:, :, ::-1]
img = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)

retval, bytes = cv2.imencode(".png", img)
encoded_image = base64.b64encode(bytes).decode("utf-8")

payload = {
    "init_images": [encoded_image],
    "prompt": "a girl with wings",
    "override_settings": {
        "sd_model_checkpoint": "MergeFlat2D_v10.safetensors [b3ae4c8c6f]",
        "CLIP_stop_at_last_layers": 2,
        "eta_noise_seed_delta": 31337
    },
    "eta": 0.667,
    "steps": 25,
    "sampler_name": "Euler a",
    "cfg_scale": 7,
    "seed": -1,
    "width": 512,
    "height": 720,
    "denoising_strength": 0.5,
    "alwayson_scripts": {
        "face editor ex": {
            "args": [
                {
                    "enabled": True,
                    "workflow": "{\n \"face_detector\": \"lbpcascade_animeface\",\n \"rules\": {\n \"then\": {\n \"face_processor\": \"img2img\",\n \"mask_generator\": {\n \"name\": \"BiSeNet\",\n \"params\": {\n \"fallback_ratio\": 0.3\n }\n }\n }\n }\n}",
                    "prompt_for_face": "smile", 
                    "show_intermediate_steps": True
                }
            ]
        },
        "controlnet": {
            "args": [
                {
                    "enabled": True,
                    "module": "canny",
                    "model": "control_v11p_sd15_canny [d14c016b]",
                    "control_mode": 1,
                    "pixel_perfect": True
                }
            ]
        }
    }
}

response = requests.post(url=f"{url}/sdapi/v1/img2img", json=payload)
print(response.status_code)

r = response.json()
for i, result in enumerate(r["images"]):
    image = Image.open(io.BytesIO(base64.b64decode(result.split(",", 1)[0])))
    image.save(f"output{i}.png")
```

**Comment by Matrixgo** [comment]
@ototadana 
Thank you again for your detailed answer! I tried it on another PC with new code version of SD and various extensions. The UI and API calls were as you mentioned, and the facial repair was normal and running very stably. 

Perhaps my previous PC's SD and face-editor version were too old, I pulled the code about six months ago, and I just tried to upgrade the face-editor separately but reported many errors. I will upgrade the SD and all extensions as a whole later.

In addition, I accidentally found that as long as the face can be detected, it can be repaired very well. Therefore, adjusting the parameter Face detection confidence from 0.97 to the lowest value of 0.7 alone can also repair the face, but it is not as stable as adjusting Workflow.

Finally, I sincerely thank you again for your timely and detailed response. This has been very helpful to me. face-editor is the lightest and most effective facial repair extension among all!

Thank you very much for your contribution!
